### PR TITLE
Drop support for legacy pipeline

### PR DESCRIPTION
**Pull request by samsondav**
Do not merge until after 0.10.11 is cut

### CODE REVIEW

**Review by jkongie — CHANGES_REQUESTED** [review]
Looks good. No major issues just some minor naming changes and a few questions.

We should look at updating any var which references jobv2 and change its name to job.

File: core/internal/cltest/mocks.go [hunk 1]
```diff
@@ -145,22 +138,23 @@ type InstanceAppFactory struct {
...
-func (s seededAppFactory) NewApplication(config *config.Config, onConnectCallbacks ...func(chainlink.Application)) (chainlink.Application, error) {
+func (s seededAppFactory) NewApplication(config *config.Config) (chainlink.Application, error) {
 	return noopStopApplication{s.Application}, nil
 }
 
 type noopStopApplication struct {
 	chainlink.Application
 }
 
+// FIXME: Why bother with this wrapper?
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/internal/cltest/mocks.go:157]: Yes but not in this PR, it's already enormous enough as it stands.

File: core/internal/features_test.go [hunk 2]
```diff
@@ -1574,192 +311,6 @@ func setupMultiWordContracts(t *testing.T) (*bind.TransactOpts, common.Address,
...
-	assert.True(t, usd.Cmp(haveUsd) == 0)
-	haveEur, err := consumer.EurInt(nil)
-	require.NoError(t, err)
-	assert.True(t, eur.Cmp(haveEur) == 0)
-	haveJpy, err := consumer.JpyInt(nil)
-	require.NoError(t, err)
-	assert.True(t, jpy.Cmp(haveJpy) == 0)
-}
-
-func TestIntegration_MultiwordV2(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/internal/features_test.go:1677]: Good catch

File: core/services/postgres/queries.go [hunk 3]
```diff
@@ -30,50 +30,6 @@ func Batch(cb BatchFunc) error {
 	}
 }
 
-// BulkDeleteRuns removes JobRuns and their related records: TaskRuns and
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/services/postgres/queries.go:33]: I don't know :)

File: core/store/migrations/0053_remove_legacy_pipeline.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
+package migrations
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+const up53 = `
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/store/migrations/0053_remove_legacy_pipeline.go:8]: I'll add a warning and note to take a backup before upgrad

File: core/store/migrations/migrate_test.go [hunk 5]
```diff
@@ -8,11 +8,9 @@ import (
```
[the remark this answers is not in the document]
Comment (samsondav) [reply core/store/migrations/migrate_test.go:8]: 🪓

**Review by jkongie — DISMISSED** [review]

**Review by archseer — APPROVED** [review]

**Review by PiotrTrzpil — APPROVED** [review]

**Review by jkongie — APPROVED** [review]

**Pull request merged**